### PR TITLE
azure - add remove tag action

### PR DESCRIPTION
**Pull request by erwelch**
Add functionality to remove tags from azure resources and resource groups.

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/actions.py [hunk 1]
```diff
@@ -78,22 +97,46 @@ def process(self, resources):
...
+            - name: azure-remove-tag-resourcegroups
+              resource: azure.resourcegroup
+              description: |
+                Remove tag for all existing resource groups with a key such as Environment
+              actions:
+               - type: tag
+                 tag: Environment
+    """
+    schema = utils.type_schema(
+        'untag', aliases=('unmark', 'remove-tag'),
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions.py]: I confirmed with @kapilt that we didn't need these aliases. (It was for backward compatibility)

File: tools/c7n_azure/c7n_azure/query.py [hunk 2]
```diff
@@ -75,6 +75,9 @@ def __new__(cls, name, parents, attrs):
             # All ARM resources will have tag support;
             # however, classic resources may not have support
             actions.register('tag', Tag)
+            actions.register('untag', RemoveTag)
+            actions.register('remove-tag', RemoveTag)
+            actions.register('unmark', RemoveTag)
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/query.py]: See earlier comment.

File: tools/c7n_azure/c7n_azure/actions.py (hunk 1 above)
Comment (erwelch) [reply tools/c7n_azure/c7n_azure/actions.py]: Oh cool, will delete!

File: tools/c7n_azure/c7n_azure/actions.py [hunk 3]
```diff
@@ -29,6 +29,25 @@ def utcnow():
...
+            resource['name'],
+            params_patch,
+        )
+    # other Azure resources
+    else:
+        az_resource = GenericResource.deserialize(resource)
+        api_version = session.resource_api_version(az_resource.id)
+        az_resource.tags = tags
+
+        client.resources.create_or_update_by_id(resource['id'], api_version, az_resource)
```
[the remark this answers is not in the document]
Comment (erwelch) [reply tools/c7n_azure/c7n_azure/actions.py]: I have the most recent version (1.2.2 : https://pypi.org/project/azure-mgmt-resource/), but seems like it fails on that method with the error that it doesn't exist. The method is in the docs, however, so it's confusing.

**Pull request merged**